src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -893,16 +893,17 @@ private static bool IsReceiverUniqueInstance(TExpressionSyntax receiver, Semanti
                 (receiverSymbol.IsKind(SymbolKind.Field) && ((IFieldSymbol)receiverSymbol).IsReadOnly);
         }
 
+        protected abstract bool HasArgumentList(TExpressionSyntax expression);
         protected abstract ImmutableArray<TArgumentSyntax> GetArguments(TExpressionSyntax expression);
         protected abstract TExpressionSyntax GetReceiver(TExpressionSyntax expression);
 
         private bool SymbolsHaveCompatibleParameterLists(ISymbol originalSymbol, ISymbol newSymbol, TExpressionSyntax originalInvocation)
         {
             if (originalSymbol.IsKind(SymbolKind.Method) || originalSymbol.IsIndexer())
             {
-                var specifiedArguments = GetArguments(originalInvocation);
-                if (specifiedArguments != null)
+                if (HasArgumentList(originalInvocation))
                 {
+                    var specifiedArguments = GetArguments(originalInvocation);
```

Comment:
what's the benefit of this pattern over hte original?  in the original, we just got hte args, then if we had any we used them.  now we need to check (which involves getting the args), then we get the args again.

Reply:
`GetArguments` never returns null. It will return an Empty Array in both cases where there are no Arguments and where there is no ArgumentList. This became an issue once this method was being called with Expressions that were lacking a ArgumentList but where the analyzer found symbols with Parameters. I added `HasArgumentList` so I could correctly identify the lack of an ArgumentList which seems to be the original intent of the null check. The compatible parameters list check will be performed though on the following pass when the expression contains the argument list.

Reply:
>  I added HasArgumentList so I could correctly identify the lack of an ArgumentList which seems to be the original intent of the null check.

I see.  So the way i would prefer this would simply be if GetArguments returned a default ImmutableArray, and you could do ```if (!arguments.IsDefault)```

Reply:
Thanks. I didn't realize `ImmutableArray` was a struct. =)